packages/trader/src/Stores/Modules/Trading/trade-store.ts

```diff
@@ -500,6 +500,7 @@ export default class TradeStore extends BaseStore {
                 if (date) {
                     this.expiry_date = date;
                 }
+                this.has_cancellation = false;
```

Comment:
@henry-deriv unfortunately, i noticed that your change is changing the behavior on Synthetics compared to production. Currently in production, when Deal cancellation is enabled on one Synthetic asset and you switch to another Synthetic, Deal cancellation remains enabled. In the test link, when Deal cancellation is enabled on one Synthetic asset and you switch to another Synthetic, Deal cancellation gets disabled. Please check🙏🏻